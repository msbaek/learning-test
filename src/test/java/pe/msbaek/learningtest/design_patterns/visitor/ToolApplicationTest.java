package pe.msbaek.learningtest.design_patterns.visitor;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.util.List;

class ToolApplicationTest {
    private ByteArrayOutputStream outputCaptor = new ByteArrayOutputStream();
    private PrintStream stdout;

    @BeforeEach
    void setUp() {
        // replace System.out with ByteArrayOutputStream
        stdout = System.out;
        System.setOut(new PrintStream(outputCaptor));
    }

    @AfterEach
    void tearDown() {
        // restore System.out
        System.setOut(stdout);
    }

    @DisplayName("3가지 타입의 리소스 파일에 대해서 extract2txt를 호출한다")
    @Test
    void main() {
        List<ResourceFile> resourceFiles = List.of(
                new PdfFile("a.pdf"),
                new WordFile("b.doc"),
                new PPTFile("c.ppt")
        );

        Extractor extractor = new Extractor();
        Compressor compressor = new Compressor();
        for (ResourceFile resourceFile : resourceFiles) {
            resourceFile.accept(extractor);
            resourceFile.accept(compressor);
        }
        Approvals.verify(outputCaptor.toString());
    }
}