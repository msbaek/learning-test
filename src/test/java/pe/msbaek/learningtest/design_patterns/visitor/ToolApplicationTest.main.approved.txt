Extract PDF.
Compress PDF.
Extract WORD.
Compress WORD.
Extract PPT.
Compress PPT.
